internal/bminventory/inventory.go

```diff
@@ -1559,6 +1560,60 @@ func (b *bareMetalInventory) GetClusterDefaultConfig(_ context.Context, _ instal
 	return installer.NewGetClusterDefaultConfigOK().WithPayload(&body)
 }
 
+func (b *bareMetalInventory) TransformClusterToDay2Internal(ctx context.Context, clusterID strfmt.UUID) (*common.Cluster, error) {
+	log := logutil.FromContext(ctx, b.log)
+	log.Infof("transforming day1 cluster %s into day2 cluster", clusterID)
+
+	var cluster *common.Cluster
+	var err error
+
+	txSuccess := false
+	tx := b.db.Begin()
+	tx = transaction.AddForUpdateQueryOption(tx)
+	defer func() {
+		if !txSuccess {
+			log.Error("transforming day1 cluster into day2 cluster has failed")
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			log.Error("transforming day1 cluster into day2 cluster has failed")
+			tx.Rollback()
+		}
+	}()
+
+	if tx.Error != nil {
+		log.WithError(tx.Error).Errorf("failed to start db transaction")
+		return nil, common.NewApiError(http.StatusInternalServerError, errors.New("DB error, failed to commit transaction"))
+	}
+
+	if cluster, err = common.GetClusterFromDB(tx, clusterID, common.UseEagerLoading); err != nil {
+		log.WithError(err).Errorf("failed to find cluster %s", clusterID)
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, common.NewApiError(http.StatusNotFound, err)
+		}
+		return nil, common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	apiVipDnsname := fmt.Sprintf("api.%s.%s", cluster.Name, cluster.BaseDNSDomain)
+	dbReply := b.db.Model(&common.Cluster{}).Where("id = ?", cluster.ID.String()).
```

Comment:
i think that this logic belong to cluster api.

Reply:
Done